community/values/src/main/java/org/neo4j/values/storable/LocalTimeValue.java

```diff
@@ -96,40 +105,56 @@ public static LocalTimeValue truncate(
         throw new UnsupportedOperationException( "not implemented" );
     }
 
-    static StructureBuilder<AnyValue,LocalTimeValue> builder( Supplier<ZoneId> defaultZone )
+    static final LocalTime DEFAULT_LOCAL_TIME = LocalTime.of( Field.hour.defaultValue, Field.minute.defaultValue );
+
+    static TimeValue.TimeBuilder<LocalTimeValue> builder( Supplier<ZoneId> defaultZone )
     {
-        return new TimeValue.TimeBuilder<AnyValue,LocalTimeValue>()
+        return new TimeValue.TimeBuilder<LocalTimeValue>( defaultZone )
         {
             @Override
-            protected ZoneId timezone( AnyValue timezone )
+            protected boolean supportsTimeZone()
             {
-                return timezone == null ? defaultZone.get() : timezoneOf( timezone );
+                return false;
             }
 
             @Override
-            protected LocalTimeValue selectTime( AnyValue temporal )
+            public LocalTimeValue buildInternal()
             {
-                throw new UnsupportedOperationException( "not implemented" );
+                LocalTime result;
+                if ( fields.containsKey( Field.time ) )
+                {
+                    AnyValue time = fields.get( Field.time );
+                    if ( !(time instanceof TemporalValue) )
```

Comment:
I agree the implementations of `buildInternal()` are very similar. Since they all have slightly different special cases, I find it hard to actually come up with a good way to generalize this. I am open for suggestions, though!